core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2086,9 +2086,9 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     // authorize for the transactionalId and the consumer group. Note that we skip producerId authorization
     // since it is implied by transactionalId authorization
-    if (!authorize(request.session, Write, Resource(TransactionalId, txnOffsetCommitRequest.transactionalId, LITERAL)))
+    if (!authorize(request.session, Write, Resource(TransactionalId, txnOffsetCommitRequest.data.transactionalId, LITERAL)))
```

Comment:
Posting this here for lack of an alternative. We have the following line under `handleOffsetCommitRequest`.
```scala
            leaderEpoch = Optional.ofNullable[Integer](partitionData.committedLeaderEpoch),
```
This seems broken because `committedLeaderEpoch` is not nullable. Can we fix this here?